controllers/storagecluster/blackbox_exporter.go

```diff
@@ -0,0 +1,563 @@
+				return err
+			}
+		}
+		actual.Data = desired.Data
+		return nil
+	})
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		r.Log.Error(err, "Failed to create/update ConfigMap for Blackbox Exporter")
+		return err
+	}
+
+	r.Log.Info("ConfigMap for Blackbox Exporter is ready")
+	return nil
+}
+
+func (r *StorageClusterReconciler) getNodeIPs(ctx context.Context) ([]string, error) {
+	nodes := &corev1.NodeList{}
+	err := r.Client.List(ctx, nodes, client.MatchingLabels{defaults.NodeAffinityKey: ""})
+	if err != nil {
+		return nil, err
+	}
+
+	var ips []string
+	for _, node := range nodes.Items {
+		for _, addr := range node.Status.Addresses {
+			if addr.Type == corev1.NodeInternalIP {
+				ips = append(ips, addr.Address)
+				break
+			}
+		}
+	}
+	return ips, nil
+}
+
+// createBlackboxDeployment deploys the Blackbox Exporter
+func (r *StorageClusterReconciler) createBlackboxDeployment(ctx context.Context, instance *ocsv1.StorageCluster) error {
+	image := os.Getenv("BLACKBOX_EXPORTER_IMAGE")
+	if image == "" {
+		image = DefaultBlackboxExporterImage
+		r.Log.Info("BLACKBOX_EXPORTER_IMAGE env var not set; using default image", "image", image)
+	}
+	desired := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      blackboxExporterName,
+			Namespace: instance.Namespace,
+			Labels:    blackboxExporterLabels,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: ptr.To(int32(1)),
+			Selector: &metav1.LabelSelector{MatchLabels: blackboxExporterLabels},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: blackboxExporterLabels,
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: blackboxServiceAccount,
+					SecurityContext: &corev1.PodSecurityContext{
+						Sysctls: []corev1.Sysctl{
+							{
+								Name:  "net.ipv4.ping_group_range",
+								Value: "0 2147483647",
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  blackboxExporterName,
+							Image: image,
+							Args: []string{
+								"--config.file=/etc/blackbox_exporter/config.yml",
+								"--web.listen-address=:9115",
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          portBlackboxHTTP,
+									ContainerPort: 9115,
+									Protocol:      corev1.ProtocolTCP,
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "config",
+									MountPath: "/etc/blackbox_exporter",
+									ReadOnly:  true,
+								},
+							},
+							SecurityContext: &corev1.SecurityContext{
+								RunAsUser:    ptr.To(int64(65534)),
+								RunAsNonRoot: ptr.To(true),
+								Capabilities: &corev1.Capabilities{
+									Add: []corev1.Capability{"NET_RAW"},
+								},
+								AllowPrivilegeEscalation: ptr.To(false),
+								SeccompProfile: &corev1.SeccompProfile{
+									Type: corev1.SeccompProfileTypeRuntimeDefault,
+								},
+							},
+							Resources: getDaemonResources(blackboxExporterName, instance),
```

Comment:
this function gets called with "odf-blackbox-exporter" but lookup will fail, as the resource.go has "blackbox-exporter"